### PR TITLE
GitHub Actions workflows for release automation

### PR DESCRIPTION
**Pull request by MartinPetkov**
Docs:
* https://docs.github.com/en/actions/configuring-and-managing-workflows/configuring-and-managing-workflow-files-and-runs
* https://docs.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token
* https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on
* https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsrun
* https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet
* https://github.com/github/hub#github-actions

### CODE REVIEW

File: .github/workflows/release-binaries.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '^1.14'
+    - run: go version
```
Comment (xingao267) [inline .github/workflows/release-binaries.yaml:37]: is the - in before run expected?

File: .github/workflows/release-binaries.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    - name: Build Go binaries
+      shell: bash
+      run: ./build/build-binaries.sh -v "$(echo '${{ github.ref }}' | sed -e 's|^refs/tags/||')"
+
+    # From https://github.com/github/hub#github-actions
+    - name: Create GitHub Release
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (xingao267) [inline .github/workflows/release-binaries.yaml:46]: whose token is this?

Comment (MartinPetkov) [reply .github/workflows/release-binaries.yaml:46]: See https://docs.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token:
"GitHub automatically creates a GITHUB_TOKEN secret to use in your workflow. You can use the GITHUB_TOKEN to authenticate in a workflow run.

Before each job begins, GitHub fetches an installation access token for the job. The token expires when the job is finished."

File: .github/workflows/release-binaries.yaml (hunk 1 above)
Comment (MartinPetkov) [reply .github/workflows/release-binaries.yaml:37]: Yes. This is technically a step without a name, that runs directly on the runner (i.e. not in a Docker container). See https://docs.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow#choosing-a-runner.

Comment (MartinPetkov) [reply .github/workflows/release-binaries.yaml:37]: And this: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsrun

Comment (MartinPetkov) [reply .github/workflows/release-binaries.yaml:37]: The action step (`uses: actions/setup-go@v2`) is the equivalent of running docker run without an explicit entrypoint, it uses the action's default command (in this case it's a bunch of commands to download Go and configure the PATH). That's why it doesn't have a "run".

**Comment by umairidris** [comment]
LGTM, will let Xin approve

**Pull request merged**